enhancements/hypershift/topology-aware-scheduling/topology-aware-scheduling.md

```diff
@@ -0,0 +1,256 @@
+
+### Handle Node Grouping
+On HyperShift NROP maps the nodes of the hosted-cluster per NodePool.
+This is the equivalent of MachineConfigPool (MCP) on OCP.
+Each node has a label with a value of the NodePool it belongs to,
+so NROP iterates over the list of nodes and builds a map of [NodePoolName] []*corev1.Nodes.
+It is suggested to use NodeSelector instead of MachineConfigPoolSelector to have a unified API for both HCP and OCP.
+A validation is necessary to prevent users from using both fields, i.e., MCP and node selectors.  
+
+### Workflow Description
+
+The deployment and usage of NROP per-se, as a day2 operator,
+on HCP is similar to the deployment and usage on OCP. 
+No major changes are necessary on that front.
+
+### API Extensions
+
+Extend the [NUMAResourceOperator](https://github.com/openshift-kni/numaresources-operator/blob/0beeb62c7c0182ba5a2963ccd658bd0f6af44ebd/api/numaresourcesoperator/v1/numaresourcesoperator_types.go#L164) CR to contain NodeSelector filed.
+`spec.NodeGroup[].NodeSelector`
+User can choose the group of node by the NodeSelector and to have a unified API for OCP and HCP.
+
+For example:
+
+  OpenShift:
+
+  `spec.NodeGroup[].NodeSelector = "node-role.kubernetes.io/worker="`
+  
+  HyperShift:
+
+  `spec.NodeGroup[].NodeSelector = "hypershift.openshift.io/nodePool=hostedcluster02"`
+
+### Topology Considerations
+
+#### Hypershift / Hosted Control Planes
+
+This proposal aims for Hypershift / Hosted Control Planes alone. 
+
+#### Standalone Clusters
+
+* N/A
+
+#### Single-node Deployments or MicroShift
+
+* N/A
+
+### Implementation Details/Notes/Constraints
+
+#### Expose KubeletConfig to Hosted-Clusters
+
+At first, when a user associates a KubeletConfig's ConfigMap to the NodePool,
+the NodePool controller mirrors the ConfigMap to the control-plane namespace.
+Similar to: https://github.com/openshift/hypershift/pull/4150
+The KubeletConfig's ConfigMap NamespacedName convention, created by the NodePool controller is:
+`<control-plane-namespace>/<name+node-pool>`
+* `control-plane-namespace` = name of the control-plane namespace
+* `name` = name of the ConfigMap as provided by the user in the hosted-cluster namespace.
+* `node-pool` = name of the NodePool on which the ConfigMap was referenced.
+
+To make sure the final name is valid,
+it's given to the https://github.com/openshift/hypershift/blob/ffa435f71e1a037174e29d31a2b80d569c940d57/support/util/route.go#L39 
+function so that 
+```
+finalName := ShortenName(name, NodePoolName)
+```
+In addition, the NodePool controller creates the KubeletConfig ConfigMap with the following labels:
+* `hypershift.openshift.io/kubeletconfig-config` : `""`
+* `hypershift.openshift.io/nodePool` : `NodePool` API name where the KubeletConfig has referenced
+
+Then, HCCO picks it up from the control-plane namespace, 
+by listing ConfigMaps with the labels above, and copies it to the hosted-cluster.
+
+The KubeletConfig's ConfigMap NamespacedName convention, created by HCCO is:
```

Comment:
Can you include a sentence to specify changes to these kubeletconfig running on the guest cluster need to be blocked by [hypershift validating admission policy](https://kubernetes.io/docs/reference/access-authn-authz/validating-admission-policy/)

Reply:
Usually it's the operator, HCCO in this case, responsibility to watch the objects it crated, and making sure to put them back in their desired state in case of tampering, isn't that so?

Reply:
right but beside's that, we have now pre persistence protection so there's no chance from anyother than the hcco to succeed on update requests https://github.com/openshift/hypershift/pull/4303